arch/arm64/src/imx9/imx9_usbdev.c

```diff
@@ -776,8 +776,9 @@ static inline void imx9_writedtd(struct imx9_dtd_s *dtd,
 
   up_clean_dcache((uintptr_t)dtd,
                   (uintptr_t)dtd + sizeof(struct imx9_dtd_s));
-  up_clean_dcache((uintptr_t)data,
-                  (uintptr_t)data + nbytes);
+
+  up_invalidate_dcache((uintptr_t)dtd->buffer0,
```

Comment:
This * 3 is.... fishy

Reply:
I'd assume you actually need to invalidate the buffer when it is allocated for receive, and again after the receive is complete.